### PR TITLE
[Serialization] Only allow loading modules during import resolution

### PR DESCRIPTION
**Pull request by jrose-apple**
Very early groundwork for private imports. Should not affect anything today.

### CODE REVIEW

**Comment by jrose-apple** [comment]
@swift-ci Please test

**Comment by jrose-apple** [comment]
@swift-ci Please test source compatibility

**Comment by harlanhaskins** [comment]
Is there a way to add a test for this?

**Comment by jrose-apple** [comment]
Hm. Yeah, maybe I can do something evil with the C preprocessor.

**Review by harlanhaskins — APPROVED** [review]

File: include/swift/Serialization/ModuleFile.h [hunk 1]
```diff
@@ -857,7 +858,7 @@ class ModuleFile
   ///
   /// If the name matches the name of the current module, a shadowed module
   /// is loaded instead.
-  ModuleDecl *getModule(ArrayRef<Identifier> name);
+  ModuleDecl *getModule(ArrayRef<Identifier> name, bool allowLoading = false);
```
Comment (slavapestov) [inline include/swift/Serialization/ModuleFile.h]: Do we ever parametrize over this or should it just be a separate function?

**Review by slavapestov — APPROVED** [review]

**Comment by jrose-apple** [comment]
@harlanhaskins How's this test look?

@swift-ci Please test

**Pull request merged**